ethereum/contracts/ERC20AtomicSwapper.sol

```diff
@@ -0,0 +1,235 @@
+pragma solidity ^0.5.8;
+
+interface ERC20 {
+    function totalSupply() external view returns (uint);
+    function balanceOf(address who) external view returns (uint);
+    function transfer(address to, uint value) external returns (bool);
+    function allowance(address owner, address spender) external view returns (uint);
+    function transferFrom(address from, address to, uint value) external returns (bool);
+    function approve(address spender, uint value) external returns (bool);
+}
+
+contract ERC20AtomicSwapper {
+
+    struct Swap {
+        uint256 erc20Amount;
+        uint256 bep2Amount;
+
+        uint256 expireHeight;
+        bytes32 secretKey;
+        uint64  timestamp;
+
+        address sender;
+        address receiverAddr;
+        bytes20 BEP2Addr;
+    }
+
+    enum States {
+        INVALID,
+        OPEN,
+        COMPLETED,
+        EXPIRED
+    }
+
+    // Events
+    event SwapInitialization(address indexed _msgSender, address indexed _receiverAddr, bytes20 _BEP2Addr, uint256 _index, bytes32 _secretHashLock, uint64 _timestamp, uint256 _expireHeight, uint256 _erc20Amount, uint256 _bep2Amount);
+    event SwapExpire(address indexed _msgSender, address indexed _swapSender, bytes32 _secretHashLock);
+    event SwapCompletion(address indexed _msgSender, address indexed _receiverAddr, bytes32 _secretHashLock, bytes32 _secretKey);
```

Comment:
keep the event name consistent, for example, "SwapInit", "SwapExpire", "SwapComplete"